io.c

```diff
@@ -766,6 +769,7 @@ static int io_write(struct io_kiocb *req, const struct sqe_submit *s,
 	ret2 = call_write_iter(file, kiocb, &iter);
 	if (!force_nonblock || ret2 != -EAGAIN) {
 		io_rw_done(kiocb, ret2);
+		ret = 0;
```

Comment:
same here.

Reply:
ret is actually -EAGAIN here. In the other case set it for consistency with this code.